pkg/operator/controller/instance_controller.go

```diff
@@ -68,7 +68,8 @@ type InstanceReconciler struct {
 // +kubebuilder:rbac:groups="",resources=services;serviceaccounts,verbs=list;get;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="discovery.k8s.io",resources=endpointslices,verbs=list;get;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=nodes,verbs=list;get;watch
-// +kubebuilder:rbac:groups="",resources=pods,verbs=list;get;watch
+// +kubebuilder:rbac:groups="",resources=pods,verbs=list;get;delete;watch
```

Comment:
might be cleaner to add RBAC for deployment as a whole and not for Pods (BTW, it can be constrained to the specific CoreDNS deployment by using `resourceNames` - not sure if  it needs to be a ClusterRole or can be namespaced. kubebuilder [rbac annotations](https://book.kubebuilder.io/reference/markers/rbac) can be used to generate what we want.

Reply:
Already removed pod deletion as it is not needed.